homeassistant/components/lgthinq/switch.py

```diff
@@ -0,0 +1,197 @@
+        icon="mdi:power",
+        name="Power",
+        translation_key="operation_power",
+        property_info=PropertyInfo(
+            key=ThinQProperty.HUMIDIFIER_OPERATION_MODE,
+            value_converter=value_to_power_state_converter,
+        ),
+    ),
+}
+
+AIR_PURIFIER_FAN_SWITCH: tuple[ThinQSwitchEntityDescription, ...] = (
+    OPERATION_SWITCH_DESC[ThinQProperty.AIR_FAN_OPERATION_MODE],
+)
+AIRPURIFIER_SWITCH: tuple[ThinQSwitchEntityDescription, ...] = (
+    OPERATION_SWITCH_DESC[ThinQProperty.AIR_PURIFIER_OPERATION_MODE],
+)
+DEHUMIDIFIER_SWITCH: tuple[ThinQSwitchEntityDescription, ...] = (
+    OPERATION_SWITCH_DESC[ThinQProperty.DEHUMIDIFIER_OPERATION_MODE],
+)
+HUMIDIFIER_SWITCH: tuple[ThinQSwitchEntityDescription, ...] = (
+    OPERATION_SWITCH_DESC[ThinQProperty.HUMIDIFIER_OPERATION_MODE],
+)
+SYSTEM_BOILER_SWITCH: tuple[ThinQSwitchEntityDescription, ...] = (
+    OPERATION_SWITCH_DESC[ThinQProperty.BOILER_OPERATION_MODE],
+)
+
+DEVIE_TYPE_SWITCH_MAP = {
+    DeviceType.AIR_PURIFIER_FAN: AIR_PURIFIER_FAN_SWITCH,
+    DeviceType.AIR_PURIFIER: AIRPURIFIER_SWITCH,
+    DeviceType.DEHUMIDIFIER: DEHUMIDIFIER_SWITCH,
+    DeviceType.HUMIDIFIER: HUMIDIFIER_SWITCH,
+    DeviceType.SYSTEM_BOILER: SYSTEM_BOILER_SWITCH,
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ThinqConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up an entry for switch platform."""
+
+    @callback
+    def add_entities(
+        coordinators: Collection[DeviceDataUpdateCoordinator],
+    ) -> None:
+        """Add switch entities."""
+        for coordinator in coordinators:
+            descriptions = DEVIE_TYPE_SWITCH_MAP.get(coordinator.device_api.device_type)
+            if not isinstance(descriptions, tuple):
+                continue
+
+            entities: list[ThinQSwitchEntity] = []
+            for description in descriptions:
+                properties = create_properties(
+                    device_api=coordinator.device_api,
+                    key=description.key,
+                    children_keys=None,
+                    rw_type=PROPERTY_WRITABLE,
+                )
+                if not properties:
+                    continue
+
+                entities += [
+                    ThinQSwitchEntity(coordinator, description, prop)
+                    for prop in properties
+                ]
+
+            if entities:
+                for entity in entities:
+                    _LOGGER.debug(
+                        "[%s] Add %s:%s entity",
+                        coordinator.device_name,
+                        Platform.SWITCH,
+                        entity.entity_description.key,
+                    )
+                async_add_entities(entities)
+
+    add_entities(entry.runtime_data.values())
+
+    entry.async_on_unload(
+        async_dispatcher_connect(hass, THINQ_DEVICE_ADDED, add_entities)
+    )
```

Comment:
Let's remove the automatic device adding for now

Reply:
Do you mean below code?
---------------------------------------------
    entry.async_on_unload(
        async_dispatcher_connect(hass, THINQ_DEVICE_ADDED, add_entities)
    )
----------------------------------------------
If so,  it is removed.

Reply:
And the callback

Reply:
We already removed it.